src/app/modules/round/round.controller.js

```diff
@@ -205,10 +206,23 @@ angular
         }
       }
 
-      // TODO Add docblock
-      function cancel() {
-        // TODO implement
-        alert('to implement');
+      /**
+       * @ngdoc method
+       * @name RoundController#undo
+       * @kind function
+       * @methodOf app.round.controller:undo
+       * @description
+       * Undo the last shot
+       */
+      function undo() {
+        vm.isTappedNumberDisabled = false;
+        vm.number = null;
+
+        if (!isShotMade(vm.shotIndex)) {
+          vm.shotIndex --;
+        }
+        vm.shots[vm.shotIndex] = null;
```

Comment:
Non ti conviene rimuovere proprio l elemento in posizione 'n' anziche metterlo a null?